### ISSUE TITLE
Tensors with `require_grad=True` are not perserved in ParticleBeam initialization with `from_parameters`

### ISSUE THREAD

**Issue by cr-xu** [desy-ml/cheetah#444]
@RemiLehe found the issue that the `require_grad` are not perserved in `ParticleBeam.from_parameters` initialization. 

```python
import cheetah
incoming = cheetah.ParticleBeam.from_parameters(
    num_particles=10,
    mu_x = torch.tensor(0.1, requires_grad=True),
    mu_y = torch.tensor(0.0),
    energy = torch.tensor(1e9)
)
incoming.x, incoming.x.requires_grad
>>>(tensor([0.1000, 0.1000, 0.1000, 0.1000, 0.1000, 0.1000, 0.1000, 0.1000, 0.1000,
         0.1000]), False)
```

The issue is due to the mutivariate normal sampling in pytorch

```python
from torch.distributions.multivariate_normal import MultivariateNormal
mean = torch.tensor([0.0], requires_grad=True)
cov = torch.tensor([[1.0]])
mvn = MultivariateNormal(mean, cov)

samples = mvn.sample((10,))
samples.requires_grad
>>>False
```

Right now to perform the gradient-based calculations one needs to manually transform the ParticleBeam after initialization.

```python
import cheetah
incoming = cheetah.ParticleBeam.from_parameters(
    num_particles=10,
    mu_x = torch.tensor(0.0, requires_grad=True),
    mu_y = torch.tensor(0.0),
    energy = torch.tensor(1e9)
)
incoming.x += torch.tensor(0.1, requires_grad=True)
incoming.x, incoming.x.requires_grad
>>>(tensor([0.1000, 0.1000, 0.1000, 0.1000, 0.1000, 0.1000, 0.1000, 0.1000, 0.1000,
         0.1000], grad_fn=<SelectBackward0>),True)
```

It would be nice to have that feature built-in at the class initialization stage.

**Comment by RemiLehe** [comment]
@cr-xu Thanks for opening the issue. 

It seems that switching from `.sample` to `.rsample` fixes this:
```
import torch
from torch.distributions.multivariate_normal import MultivariateNormal
mean = torch.tensor([0.0], requires_grad=True)
cov = torch.tensor([[1.0]])
mvn = MultivariateNormal(mean, cov)

samples = mvn.rsample((10,))
samples.requires_grad
>>> True
```

**Comment by RemiLehe** [comment]
@cr-xu Do you see any issue with switching from `.sample` to `.rsample` throughout Cheetah?

**Comment by cr-xu** [comment]
Oh that's an easy fix! 
I don't expect any issues at the first glance, because it's just performing a transformation on the generated samples.
